### PR TITLE
Add exercise 2

### LINKED ISSUE #18: Second task: Print to serial, write to bluetooth
From the location puck, go to write to serial. Display serial serial in an app like PuTTY or Termite. Set up gatt attributes and write to these from phone app. Display stuff written from phone app to serial port.

### PR DESCRIPTION
**Pull request by sigvef**
This fixes #18.

Not complete yet, but please comment.

### CODE REVIEW

File: exercises/exercise-2.md [hunk 1]
```diff
@@ -3,3 +3,42 @@ layout: guide
 ---
 
 # Exercise 2
+
+In exercise 2, we are going to make a program that is capable recieving chat messages over bluetooth, and printing those messages to a terminal on your laptop using the USB connection.
```
Comment (stianjensen) [inline exercises/exercise-2.md]: receive/weird/neigh/

Comment (stianjensen) [inline exercises/exercise-2.md]: 'capable receiving'

File: exercises/exercise-2.md [hunk 2]
```diff
@@ -3,3 +3,42 @@ layout: guide
...
+
+1. **Create a new project on the [online compiler](https://developer.mbed.org/compilter) based on the [Location Puck](todo) example.**
```
Comment (stianjensen) [inline exercises/exercise-2.md]: compilter

File: exercises/exercise-2.md [hunk 3]
```diff
@@ -1,5 +1,45 @@
 ---
 layout: guide
+banner: images/ex2-banner.jpg
 ---
 
 # Exercise 2
+
+In exercise 2, we are going to make a program that is capable of receiving chat messages over bluetooth, and printing those messages to a terminal on your laptop using the USB connection.
+
```
Comment (cristeahub) [inline exercises/exercise-2.md:9]: `A real chat client!` Make some hype here?

File: exercises/exercise-2.md [hunk 4]
```diff
@@ -1,5 +1,45 @@
...
+1. **Create a new project on the [online compiler](https://developer.mbed.org/compiler) based on the [Location Puck](todo) example.**
+Remember to set a unique ID in the puck `init` function, as this will make your life much easier later on.
+You can find a guide on how to create new projects in the guides section if you get stuck.
+
+1. **Create a GATT attribute for receiving chat messages using the Puck library.**
+In order to be able to receive information over Bluetooth, we need to make a writable GATT attribute.
```
Comment (cristeahub) [inline exercises/exercise-2.md]: Explain what GATT means?

File: exercises/exercise-2.md [hunk 5]
```diff
@@ -1,5 +1,45 @@
...
+banner: images/ex2-banner.jpg
 ---
 
 # Exercise 2
+
+In exercise 2, we are going to make a program that is capable of receiving chat messages over bluetooth, and printing those messages to a terminal on your laptop using the USB connection.
+
+1. **Create a new project on the [online compiler](https://developer.mbed.org/compiler) based on the [Location Puck](todo) example.**
+Remember to set a unique ID in the puck `init` function, as this will make your life much easier later on.
+You can find a guide on how to create new projects in the guides section if you get stuck.
```
Comment (cristeahub) [inline exercises/exercise-2.md]: There is a pending prq on "how to use import libraries" that also says how one to create a new project. I guess we can either rename that guide or change the wording here?

File: exercises/exercise-2.md [hunk 6]
```diff
@@ -1,5 +1,45 @@
...
+When you have made an attribute, compiled, flashed and run your code on the development board, you should be able to see the attribute available from your smartphone using a BLE inspector app such as [B-BLE]() or [LightBlue]().
+
+1. **Add a `onDataWritten` handler to your program.**
+Now that we are able to write data to the puck over bluetooth using our cell phones, we should find something useful to do with this data.
+One cool thing to do is to print it out to a terminal on your laptop, so let's do that in the next step.
+
+1. **Install a terminal on your machine if you don't already have one.**
+For Windows, you can use termite.
+For Linux, you can use *something*.
+For OS X, you can use *something*.
```
Comment (cristeahub) [inline exercises/exercise-2.md]: CoolTerm!

File: exercises/exercise-2.md [hunk 7]
```diff
@@ -1,5 +1,45 @@
...
+1. **Start the terminal, run the program, and see if it all works!**
+You should be able to connect to the puck with your phone, write messages to your `message` GATT attribute, and see them appear in your terminal on the laptop.
+Pretty nifty.
+
+![](images/ex2-termite.png)
+
+This concludes exercise 2.
+Well done if you were able to get this far with everything working!
+If you are stuck on a step above, raise your hand and we will come help you out.
+Don't forget to check out the troubleshooting section at the bottom of this page -- maybe it will help you figure out what is the problem.
```
Comment (cristeahub) [inline exercises/exercise-2.md]: Two `-` characters or just one?

**Comment by cristeahub** [comment]
How are we doing it with tips and hints?

File: exercises/exercise-2.md (hunk 7 above)
Comment (sigvef) [inline exercises/exercise-2.md]: Looks like `--` is non-standard Markdown. I was gunning for an mdash: &mdash;.

File: exercises/exercise-2.md (hunk 4 above)
Comment (sigvef) [inline exercises/exercise-2.md]: Hopefully the slides will provide this information before ex 2.

Comment (cristeahub) [inline exercises/exercise-2.md]: Ye!

File: exercises/exercise-2.md [hunk 8]
```diff
@@ -1,5 +1,45 @@
 ---
 layout: guide
+banner: images/ex2-banner.jpg
 ---
 
 # Exercise 2
+
+In exercise 2, we are going to make a program that is capable of receiving chat messages over bluetooth, and printing those messages to a terminal on your laptop using the USB connection.
```
Comment (stianjensen) [inline exercises/exercise-2.md]: Should we try and consistently write Bluetooth with a capital B?

**Comment by sigvef** [comment]
ping

**Comment by sigvef** [comment]
Anyone have any good serial terminal suggestions for linux, btw?

**Comment by stianjensen** [comment]
This seems like 'the shit':

https://help.ubuntu.com/community/Minicom

**Comment by sigvef** [comment]
Ah yes, we used minicom in tdt4258

**Comment by sigvef** [comment]
pgin

**Comment by sigvef** [comment]
ping*

File: exercises/exercise-2.md [hunk 9]
```diff
@@ -1,5 +1,38 @@
...
+In exercise 2, we are going to make a program that is capable of receiving chat messages over Bluetooth, and printing those messages to a terminal on your laptop using the USB connection.
+
+1. **Create a new project on the [online compiler](https://developer.mbed.org/compiler) based on the [Location Puck](todo) example.**
+Remember to set a unique ID in the puck `init` function, as this will make your life much easier later on.
+You can find a guide on how to create new projects in the guides section if you get stuck.
+
+1. **Create a GATT attribute for receiving chat messages using the Puck library.**
+In order to be able to receive information over Bluetooth, we need to make a writable GATT attribute.
+Make one that is 20 bytes long (the maximum length allowed for attributes).
+When you have made an attribute, compiled, flashed and run your code on the development board, you should be able to see the attribute available from your smartphone using a BLE inspector app such as [B-BLE]() or [LightBlue]().
```
Comment (stianjensen) [inline exercises/exercise-2.md]: Missing links

File: exercises/exercise-2.md [hunk 10]
```diff
@@ -1,5 +1,38 @@
...
+
+1. **Add a `onDataWritten` handler to your program.**
+Now that we are able to write data to the puck over Bluetooth using our cell phones, we should find something useful to do with this data.
+One cool thing to do is to print it out to a terminal on your laptop, so let's do that in the next step.
+
+1. **Install a terminal on your machine if you don't already have one.**
+For Windows, you can use [Termite](http://www.compuphase.com/software/termite-3.1.exe).
+For Linux, you can use [Minicom](https://help.ubuntu.com/community/Minicom).
+For OS X, you can use [CoolTerm](http://freeware.the-meiers.org/CoolTermMac.zip).
+If you are on Windows, you will also need to [install a serial driver](http://developer.mbed.org/handbook/Windows-serial-configuration).
```
Comment (stianjensen) [inline exercises/exercise-2.md:29]: We could provide a link here to the more extensive guide on setting up serial debugging, for people to look at if they get stuck

**Comment by stianjensen** [comment]
Where will we file 'images/ex-2.png' be used?

File: exercises/exercise-2.md [hunk 11]
```diff
@@ -1,5 +1,38 @@
 ---
 layout: guide
+banner: images/ex2-banner.jpg
 ---
 
 # Exercise 2
+
+In exercise 2, we are going to make a program that is capable of receiving chat messages over Bluetooth, and printing those messages to a terminal on your laptop using the USB connection.
+
+1. **Create a new project on the [online compiler](https://developer.mbed.org/compiler) based on the [Location Puck](todo) example.**
```
Comment (cristeahub) [inline exercises/exercise-2.md]: ~~This is mostly explained in the `import a library` guide~~ Scratch that, people should know that by this point

**Comment by sigvef** [comment]
> Where will we file 'images/ex-2.png' be used?

Accidental add, I made it for exercise 1 a couple of days ago.

**Comment by sigvef** [comment]
ping

File: exercises/exercise-2.md [hunk 12]
```diff
@@ -1,5 +1,39 @@
...
+In exercise 2, we are going to make a program that is capable of receiving chat messages over Bluetooth, and printing those messages to a terminal on your laptop using the USB connection.
+
+1. **Create a new project on the [online compiler](https://developer.mbed.org/compiler) based on the Puck library.**
+Remember to set a unique ID in the puck `init` function, as this will make your life much easier later on.
+Check out the [library importing guide](guides/mbed-import.html) if you get stuck.
+
+1. **Create a GATT Characteristic for receiving chat messages using the Puck library.**
+In order to be able to receive information over Bluetooth, we need to make a writable GATT characteristic.
+Make one that is 20 bytes long (the maximum length allowed for characteristics).
+When you have made a characteristic, compiled, flashed and run your code on the development board, you should be able to see the characteristic available from your smartphone using a BLE inspector app such as [B-BLE]() or [LightBlue]().
```
Comment (stianjensen) [inline exercises/exercise-2.md]: Still missing links? o.0

Comment (sigvef) [inline exercises/exercise-2.md]: Missing link comment drowned in a different edit on the same line :/

**Comment by sigvef** [comment]
ping

**Pull request merged**